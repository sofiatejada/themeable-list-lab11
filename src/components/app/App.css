/* Dark Theme*/
.bgDark {
  background-color: black;
  color: white;
}

.dark {
  text-decoration: none;
  color: white;
  background-color: black;
  padding: 5px;
}

.hov:hover {
  background-color: white;
  color: black;
}


/* LightTheme */
.bgLight {
  background-color: white;
  color: black;
}

.light {
  text-decoration: none;
  color: black;
  background-color: white;
  padding: 5px;
}

.lhov:hover {
  background-color: black;
  color: white;
}
