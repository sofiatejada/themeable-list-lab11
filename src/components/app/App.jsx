import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import CharacterList from '../characters/CharacterList';
import CharacterDetail from '../characters/CharacterDetail';
import Header from '../characters/Header';
import styles from './App.css';
import { useTheme, useThemeUpdate } from '../../hooks/ThemeProvider';

export default function App() {
  const darkTheme = useTheme();
  const toggleTheme = useThemeUpdate();

  return (
    <BrowserRouter>
      <div 
        className={
          `${
            darkTheme ? styles.bgDark : 
              !darkTheme ? styles.bgLight : toggleTheme }`
        }>
        <Header />
        <Switch>
          <Route exact path="/" component={CharacterList} />
          <Route exact path="/:id" component={CharacterDetail} />
        </Switch>
      </div>
    </BrowserRouter>
  );
}
