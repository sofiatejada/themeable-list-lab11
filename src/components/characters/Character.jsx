import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useTheme, useThemeUpdate } from '../../hooks/ThemeProvider';
import styles from '../app/App.css';
import { BrowserRouter } from 'react-router-dom';

export default function Character({ name, image, id }) {
  const darkTheme = useTheme();
  const toggleTheme = useThemeUpdate();

  return (
    <BrowserRouter>
    <figure>
      <img src={image} alt={`image of ${name}`} />
      <Link 
        className={`
        ${darkTheme ? styles.dark : !darkTheme ? styles.light : toggleTheme}
        ${darkTheme ? styles.hov : !darkTheme ? styles.lhov : toggleTheme}`} 
        key={id} to={id}>
        {name}
      </Link>
    </figure>
    </BrowserRouter>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string
};
