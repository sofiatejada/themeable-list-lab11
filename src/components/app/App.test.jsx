import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import CharacterList from '../characters/CharacterList';
import Character from '../characters/Character';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mock from './mock.json';
import App from './App';

const server = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters/', 
    (req, res, ctx) => {
      return res(
        ctx.json(mock)
      );
    })
);

describe('character component', () => {

  it('displays a character', () => {
    render(
      <Character 
        id="1" 
        name="poop"
        image="example.png" />
    );

    const article = screen.getByRole('img', { name: 'image of poop' });
    expect(article).toMatchSnapshot();
  });
});

describe('character list component', () => {

  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of characters', async () => {
    render(
      <MemoryRouter>
        <CharacterList />
      </MemoryRouter>
    );
    
    screen.getByAltText('loading gif');
    const list = await screen.findByLabelText('character-list');
    screen.debug();
    expect(list).toMatchSnapshot();
  });

  it('has a button that toggles dark and light mode', async () => {
    const { container } = render(<App />);
    expect(container).not.toBeEmptyDOMElement();

    const button = screen.getByLabelText('button');
    const main = await screen.findByLabelText('main');

    fireEvent.click(button);
    waitFor(() =>
      expect(main.className).toEqual('bgLight'));

    fireEvent.click(button);
    waitFor(() =>
      expect(main.className).toEqual('bgDark'));

  });
});
