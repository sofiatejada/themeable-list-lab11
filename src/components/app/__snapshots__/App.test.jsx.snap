// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component displays a character 1`] = `
<img
  alt="image of poop"
  src="example.png"
/>
`;

exports[`character list component displays a list of characters 1`] = `
<ul
  aria-label="character-list"
>
  <li>
    <figure>
      <img
        alt="image of Nancy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f5c"
      >
        Nancy
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Grandpa's grandfather"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f59"
      >
        Grandpa's grandfather
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Grandpa Phil"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f52"
      >
        Grandpa Phil
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Grandpa's father"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f58"
      >
        Grandpa's father
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Phoebe Heyerdahl"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f70"
      >
        Phoebe Heyerdahl
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Eugene Horowitz"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f7e"
      >
        Eugene Horowitz
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Ray"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f82"
      >
        Ray
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Shelley"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f8b"
      >
        Shelley
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Lila Sawyer"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f76"
      >
        Lila Sawyer
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Uncle Stinky"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f7c"
      >
        Uncle Stinky
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Arnie"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f55"
      >
        Arnie
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Olga Pataki"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f66"
      >
        Olga Pataki
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Timberly Johanssen"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f6b"
      >
        Timberly Johanssen
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Jerry Berman"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f6d"
      >
        Jerry Berman
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Rhonda Wellington Lloyd"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f73"
      >
        Rhonda Wellington Lloyd
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Stinky Peterson"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f78"
      >
        Stinky Peterson
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Kyo Heyerdahl"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f71"
      >
        Kyo Heyerdahl
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Iggy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f8e"
      >
        Iggy
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Mr. Sawyer"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f77"
      >
        Mr. Sawyer
      </a>
    </figure>
  </li>
  <li>
    <figure>
      <img
        alt="image of Thaddeus \\"Curly\\" Gammelthorpe"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
      />
      <a
        class="
        light
        lhov"
        href="/5da237699734fdcb7bef8f84"
      >
        Thaddeus "Curly" Gammelthorpe
      </a>
    </figure>
  </li>
</ul>
`;
